File: Programmers/Lv2/Lv2_소수만들기.py
import itertools

def solution(nums):
    answer = 0
    sums = []
    for sum3 in itertools.combinations(nums, 3):
        sums.append(sum(sum3))

    n = max(sums)
    check = [False] * (n + 1)
    check[0] = check[1] = True
    for i in range(2, n + 1):
        if check[i] == False:
            for j in range(i * i, n + 1, i):
                check[j] = True
    for i in sums:
        if check[i] == False:
            answer += 1

    return answer

File: Programmers/Lv2/Lv2_소수만들기.cpp
﻿#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

bool isprime(int n) {
	int i;
	for (i = 2; i < n; i++) {
		if (n%i == 0)
			return false;
	}
	return true;		//소수
}

int solution(vector<int> nums) {
	int answer = 0;
	vector<int> v(nums.size());
	v[0] = 1; v[1] = 1; v[2] = 1;
	sort(v.begin(), v.end());

	do {
		int sum = 0;
		for (int i = 0; i < nums.size(); i++) {
			if (v[i] == 1)
				sum += nums[i];
		}
		if (isprime(sum))
			answer++;
	} while (next_permutation(v.begin(), v.end()));
	
	return answer;
}
